frontend/src/services/UserFunctions.ts

```diff
@@ -16,6 +16,24 @@ export async function addToUserCollection(userCredential: UserCredential): Promi
 }
 
 export const fetchAdminStatus = async () => {
-    const res = await callUserFunction("admin/checkAdminStatus", "GET");
-    return res
+    const result = await callUserFunction("admin/checkAdminStatus", "GET");
+    if (result.success) {
+        return result.data;
+    } else {
+        throw new Error(result.error || "Unable to fetch admin status");
+    }
+}
+
+export async function getUsernameByUid(uid: string) {
+    try {
+        const result = await callUserFunction(`user/username/${uid}`)
+        if (result.success) {
+            return result.data;
+        } else {
+            throw new Error(result.error || "Unable to fetch username");
+        }
+    } catch (error) {
+        console.error("Error checking uid for username:", error);
+        throw error; // rethrow the error for further handling
+    }
```

Comment:
Should change the usage of `callUserFunction` to `callFunction`.

Reply:
Amended!